### ISSUE TITLE
relationId is empty

### ISSUE THREAD

**Issue by waningflow** [catamphetamine/read-excel-file#37]
When I parse a `.xlsx` file with a sheet named `Sheet1`, I got an error `Error: Sheet #1 not found in the *.xlsx file. Available sheets: "Sheet1"`. Then I dig deeper and I found the parsed properties are like below.
![image](https://user-images.githubusercontent.com/6823964/63266160-e18b7300-c2c1-11e9-87b3-c3ead02eccbd.png)
Thus the empty `relationId` is the cause. Might you please help to figure out this problem?

**Comment by catamphetamine** [comment]
Post the excel file.

**Comment by waningflow** [comment]
A strange case is when I copy the content to a new file or save the origin file manually it will be ok.

**Comment by waningflow** [comment]
> Post the excel file.

[test.xlsx](https://bd-bucket-19.s3.amazonaws.com/test.xlsx)

**Comment by catamphetamine** [comment]
The `xl/workbook.xml` file is:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<workbook xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main" xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" xmlns:r="http://schemas.openxmlformats.org/officeDocument/2006/relationships" xmlns:x15="http://schemas.microsoft.com/office/spreadsheetml/2010/11/main" mc:Ignorable="x15">
   <fileVersion appName="xl" lastEdited="7" lowestEdited="7" rupBuild="10715" />
   <workbookPr defaultThemeVersion="166925" />
   <bookViews>
      <workbookView windowHeight="16600" windowWidth="28800" xWindow="0" yWindow="460" />
   </bookViews>
   <sheets>
      <sheet xmlns:relationships="http://schemas.openxmlformats.org/officeDocument/2006/relationships" name="Sheet1" sheetId="1" relationships:id="rId1" />
   </sheets>
   <calcPr calcId="181029" />
   <extLst>
      <ext uri="{140A7094-0E35-4892-8432-C4D2E57EDEB5}">
         <x15:workbookPr chartTrackingRefBase="1" />
      </ext>
      <ext xmlns:xcalcf="http://schemas.microsoft.com/office/spreadsheetml/2018/calcfeatures" uri="{B58B0392-4F1F-4190-BB64-5DF3571DCE5F}">
         <xcalcf:calcFeatures>
            <xcalcf:feature name="microsoft.com:RD" />
            <xcalcf:feature name="microsoft.com:Single" />
            <xcalcf:feature name="microsoft.com:FV" />
            <xcalcf:feature name="microsoft.com:CNMTM" />
         </xcalcf:calcFeatures>
      </ext>
   </extLst>
</workbook>
```

The `relationships:id="rId1"` is not parsed for some reason.
Usually XLSX files declare it as `r:id` and `xmlns:r="http://schemas.openxmlformats.org/officeDocument/2006/relationships"` on the root element.
The file you supplied does have `xmlns:r="http://schemas.openxmlformats.org/officeDocument/2006/relationships"` on the root element but also has the same namespace URL with another namespace on `<sheet/>` elements.
I guess that's the issue.

I don't know if it's a valid XML but neither `xpath` library nor the in-browser `XMLDom` are able to parse it.
So I'd assume the XML is incorrect.

**Comment by waningflow** [comment]
The excel was generated by [excelize](https://github.com/360EntSecGroup-Skylar/excelize). I found there are several differences with the format that this lib aims to parse. For example
- r:id -> relationships:id
- dimensions is not reliable
- cell's type may be `str` which this lib didn't consider
I'm not sure whether it's necessary to make it compatible with this format. If it's incorrect, why it can be opened by Excel correctly? Is there a standard of the xml format?

**Comment by catamphetamine** [comment]
> dimensions is not reliable

Yes, I did experiment with replacing `relationships:id` -> `r:id` and it threw the dimensions error so I figured the Excel is very different from what usually is being open using this library.

> I'm not sure whether it's necessary to make it compatible with this format.

This is the first case this library doesn't open an XSLX file.
It's true that the file can be opened using various Excel-like software.
I guess we'll leave it as is because I surely won't be interested in developing it any further because it has been developed for a company I work for and it works there so far without issues.

> Is there a standard of the xml format?

That's probably the most obscure thing: all this library has been developed via "trial and error" and the format seems so complicated that there's no single guide on it.
